nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -0,0 +1,371 @@
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+
+    { systemd.timers."nextcloud-cron" = {
+        wantedBy = [ "timers.target" ];
+        timerConfig.OnBootSec = "5m";
+        timerConfig.OnUnitActiveSec = "15m";
+        timerConfig.Unit = "nextcloud-cron.service";
+      };
+
+      systemd.services = {
+        "nextcloud-setup" = let
+          overrideConfig = pkgs.writeText "nextcloud-config.php" ''
+            <?php
+            $CONFIG = [
+              'apps_paths' => [
+                [ 'path' => '${cfg.home}/apps', 'url' => '/apps', 'writable' => false ],
+                [ 'path' => '${cfg.home}/store-apps', 'url' => '/store-apps', 'writable' => true ],
+              ],
+              'datadirectory' => '${cfg.home}/data',
+              'skeletondirectory' => '${cfg.skeletonDirectory}',
+              ${optionalString cfg.caching.apcu "'memcache.local' => '\\OC\\Memcache\\APCu',"}
+              'log_type' => 'syslog',
+            ];
+          '';
+          autoConfig = pkgs.writeText "nextcloud-autoconfig.php" (let acfg = cfg.autoconfig; in ''
+            <?php
+            $AUTOCONFIG = [
+              "dbtype"        => "${acfg.dbtype}",
+              "dbname"        => "${acfg.dbname}",
+              "dbuser"        => "${acfg.dbuser}",
+              "dbpass"        => "${acfg.dbpass}",
+              "dbhost"        => "${acfg.dbhost}",
+              "dbtableprefix" => "${acfg.dbtableprefix}",
+              "adminlogin"    => "${acfg.adminlogin}",
+              "adminpass"     => "${acfg.adminpass}",
+              "directory"     => "${cfg.home}/data",
+            ];
+          '');
+        in {
+          wantedBy = [ "multi-user.target" ];
+          before = [ "phpfpm-nextcloud.service" ];
+          script = ''
+            chmod og+x ${cfg.home}
+            ln -sf ${pkgs.nextcloud}/apps ${cfg.home}/
+            mkdir -p ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps
+            ln -sf ${overrideConfig} ${cfg.home}/config/override.config.php
+
+            # Do not create autoconfig if already installed
+            [ -e ${cfg.home}/config/config.php ] && exit 0
+
+            ln -sf ${autoConfig} ${cfg.home}/config/autoconfig.php
+            chown -R nextcloud:nginx ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps
+          '';
+          serviceConfig.Type = "oneshot";
+        };
+        "nextcloud-cron" = {
+          environment.NEXTCLOUD_CONFIG_DIR = "${cfg.home}/config";
+          serviceConfig.Type = "oneshot";
+          serviceConfig.User = "nextcloud";
+          serviceConfig.ExecStart = "${pkgs.php}/bin/php -f ${pkgs.nextcloud}/cron.php";
+        };
+      };
+
+      services.phpfpm = {
+        phpOptions = phpOptionsExtensions;
+        phpPackage = pkgs.php71;
+        pools.nextcloud = let
+          phpAdminValues = concatStringsSep "\n" (flip mapAttrsToList phpOptions
+            (k: v: "php_admin_value[${k}] = ${v}"));
```

Comment:
Something weird is going on here:

```

[root@cloud:~]# cat /nix/store/jkqz260cfm0335fvpyflpvdhj1hgkqjp-nextcloud-occ/bin/nextcloud-occ
#! /nix/store/czx8vkrb9jdgjyz8qfksh10vrnqa723l-bash-4.4-p23/bin/bash
cd /nix/store/wni0plnr91xlcrz2lfwr4dzlxlmb058r-nextcloud-13.0.5
exec /nix/store/bgwy6ysr8wffmy74xn9pj978gxas4n7j-sudo-1.8.23/bin/sudo -u nextcloud \
  NEXTCLOUD_CONFIG_DIR="/data/var/lib/nextcloud/config" \
  /nix/store/bmwiwi69556177ndfdg0nc6pg0j84ky1-php-7.1.21/bin/php \
  -c /nix/store/vlpanb3f5iigwkigr3ywg6zzryywkwng-php.ini\
  occ $*

[root@cloud:~]# cat /nix/store/vlpanb3f5iigwkigr3ywg6zzryywkwng-php.ini
extension=/nix/store/vfkv108a0zya61b4gxxyym4jj9fr9pyk-php-apcu-5.1.11/lib/php/extensions/apcu.so


zend_extension = opcache.so
opcache.enable = 1
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
k = v
```
It could be related to this line?

Reply:
Actually, it's due the literal `k = v` here: https://github.com/NixOS/nixpkgs/pull/44994/files#diff-fc6e7fa5e8f92a630f0738fa2fc4e9faR20 :sweat_smile:

Reply:
Whoops, nice find!